core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -70,10 +70,27 @@ class Ruleset {
     var victories = LinkedHashMap<String, Victory>()
     var cityStateTypes = LinkedHashMap<String, CityStateType>()
 
-    val greatGeneralUnits by lazy { 
+    val greatGeneralUnits by lazy {
         units.values.filter { it.hasUnique(UniqueType.GreatPersonFromCombat, StateForConditionals.IgnoreConditionals) }
     }
 
+    /** Contains all happiness levels that moving *from* them, to one *below* them, can change uniques that apply */
+    val allHappinessLevelsThatAffectUniques by lazy {
+        sequence {
+            for (rulesetObject in (units.values + terrains.values + buildings.values + technologies.values + eras.values
```

Comment:
Something tells me we should have a helper function/lazy for this in case it comes up again that we're looking through all objects. Half surprised we don't already have one